### PR TITLE
Do not verify pod Swift version from integrated targets if integration is disabled.

### PR DESCRIPTION
**Pull request by dnkoutso**
closes https://github.com/CocoaPods/CocoaPods/issues/8653

### CODE REVIEW

**Comment by CocoaPodsBarista** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please include a CHANGELOG entry to credit yourself! <br />
You can find it at <a href="https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>

Here's an example of your CHANGELOG entry:

```markdown
* Do not verify pod Swift version if integration is disabled  
  [dnkoutso](https://github.com/dnkoutso)
  [#issue_number](https://github.com/CocoaPods/CocoaPods/issues/issue_number)
```

*note*: There are two invisible spaces after the entry's text.


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: lib/cocoapods/installer/xcode/target_validator.rb [hunk 1]
```diff
@@ -34,7 +36,7 @@ def initialize(aggregate_targets, pod_targets)
         def validate!
           verify_no_duplicate_framework_and_library_names
           verify_no_static_framework_transitive_dependencies
-          verify_swift_pods_swift_version
+          verify_swift_pods_swift_version if installation_options.integrate_targets?
```
Comment (segiddins) [inline lib/cocoapods/installer/xcode/target_validator.rb]: can we also add a test for what swift version should be used when integrating targets is disabled?

Comment (dnkoutso) [reply lib/cocoapods/installer/xcode/target_validator.rb]: hm good point......what version should it be set to?

Comment (segiddins) [reply lib/cocoapods/installer/xcode/target_validator.rb]: The highest version the pod supports?

Comment (dnkoutso) [reply lib/cocoapods/installer/xcode/target_validator.rb]: and what if the pod does not specify `swift_version`?

Comment (segiddins) [reply lib/cocoapods/installer/xcode/target_validator.rb]: Then we have to error, right?

Comment (dnkoutso) [reply lib/cocoapods/installer/xcode/target_validator.rb]: This class does not _set_ the swift version for the target, only validates it. I'd argue none of these checks should happen if integration is off.

Comment (amorde) [reply lib/cocoapods/installer/xcode/target_validator.rb]: Wouldn't that be a reason to fail validation?

Comment (segiddins) [reply lib/cocoapods/installer/xcode/target_validator.rb]: I think so, @amorde

File: spec/unit/installer/xcode/target_validator_spec.rb [hunk 2]
```diff
@@ -42,8 +42,8 @@ def create_validator(sandbox, podfile, lockfile)
             config.repos_dir = fixture_path + 'spec-repos'
             podfile = Pod::Podfile.new do
               platform :ios, '8.0'
-              install! 'cocoapods', :integrate_targets => false
-              project 'SampleProject/SampleProject'
+              install! 'cocoapods', :integrate_targets => true
```
Comment (dnkoutso) [inline spec/unit/installer/xcode/target_validator_spec.rb]: all of these tests were "wrong" as in they were never taking into consideration `:integrate_targets => false` above and the project was never properly loaded. This is now fixed.

File: spec/unit/installer/xcode/target_validator_spec.rb [hunk 3]
```diff
@@ -322,69 +321,82 @@ def create_validator(sandbox, podfile, lockfile)
             'the author or set the `SWIFT_VERSION` attribute in at least one of the targets that integrate this pod.'
           end
 
-          it 'does not raise when targets integrate the same pod but only one of the targets is a swift target' do
+          it 'does not raise when integrating targets is disabled' do
```
Comment (dnkoutso) [inline spec/unit/installer/xcode/target_validator_spec.rb]: diff is weird but this is the new test here that explicitly states `integrate_targets => false`

**Comment by dnkoutso** [comment]
After much work and understanding this is not the right direction we want here.

**Pull request closed without merging**